### PR TITLE
Add optics-agent typings

### PR DESCRIPTION
**Pull request by Crevil**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [x] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

### CODE REVIEW

File: optics-agent/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
```
Comment (andy-ms) [inline optics-agent/tsconfig.json]: Use `npm run new-package -- optics-agent` to generate this file and `tslint.json`. (You'll have to rename the folder to `optics-agent-old`, the move your `index.d.ts` and `optics-agent-tests.ts` back.)

File: optics-agent/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  }
+}
+
+export function configureAgent(options: OpticsAgent.Options): OpticsAgent.Agent;
+export function instrumentSchema(schema: GraphQLSchema): void;
+export function middleware(): (req: Request, res: Response, next?: any) => void;
+export function instrumentHapiServer(server: Server): void;
+export function context(req: Request): any;
+
+export default {
```
Comment (andy-ms) [inline optics-agent/index.d.ts]: I find it surprising that the library would do this, but I've verified that it is indeed accurate.

File: optics-agent/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+// Type definitions for optics-agent 1.0
+// Project: https://github.com/apollostack/optics-agent-js
+// Definitions by: Crevil <https://github.com/crevil>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { GraphQLSchema } from "graphql";
+import { Request, Response } from "express";
+import { Server } from "hapi";
+
+declare namespace OpticsAgent {
```
Comment (andy-ms) [inline optics-agent/index.d.ts]: Don't need the namespace. You can just directly export these.

File: optics-agent/index.d.ts (hunk 2 above)
Comment (Crevil) [inline optics-agent/index.d.ts]: As was I. Thanks for verifying it.

**Comment by Crevil** [comment]
@andy-ms I've changed the requested parts. Let me know if anything else should be done before merging.

**Comment by andy-ms** [comment]
Typings look good, just needs `"noImplicitThis": true` in `tsconfig.json` to make tests pass.

**Comment by Crevil** [comment]
@andy-ms That's added.

**Comment by Crevil** [comment]
@andy-ms Is it possible to depend on the [`typed-graphql`](https://github.com/nitintutlani/typed-graphql) implementation of `graphql` typings instead of `@types/graphql`?

**Pull request merged**